packages/devtools_app/test/network/network_request_inspector_test.dart

```diff
@@ -129,5 +130,197 @@ void main() {
     });
+
+    group('HttpResponseTrailingDropDown', () {
+      testWidgets(
+          'drop down value should update when response view type changes',
+          (tester) async {
+        NetworkResponseViewType? getCurrentDropDownValue() {
+          final RoundedDropDownButton<NetworkResponseViewType> dropDownWidget =
+              find
+                  .byType(RoundedDropDownButton<NetworkResponseViewType>)
+                  .evaluate()
+                  .first
+                  .widget as RoundedDropDownButton<NetworkResponseViewType>;
+          return dropDownWidget.value;
+        }
+
+        final currentResponseViewType = controller.currentResponseViewType;
+        final requestsNotifier = controller.requests;
+        await controller.startRecording();
+
+        // Load the network request.
+        await controller.networkService.refreshNetworkData();
+        expect(requestsNotifier.value.requests.length, equals(1));
+
+        // Select the request in the network request list.
+        final data = requestsNotifier.value.requests.first;
+
+        // Matches Drop Down value with currentResponseViewType
+        void checkDropDownValue() {
+          final currentDropDownValue = getCurrentDropDownValue();
+          expect(currentDropDownValue, equals(currentResponseViewType.value));
+        }
+
+        await tester.pumpWidget(
+          wrapWithControllers(
+            HttpResponseTrailingDropDown(
+              data as DartIOHttpRequestData,
+              currentResponseViewType: currentResponseViewType,
+              onChanged: (value) {
+                controller.setResponseViewType = value;
+              },
+            ),
+            debugger: createMockDebuggerControllerWithDefaults(),
+          ),
+        );
+
+        controller.setResponseViewType = NetworkResponseViewType.json;
+        await tester.pumpAndSettle();
+        checkDropDownValue();
+
+        controller.setResponseViewType = NetworkResponseViewType.text;
+        await tester.pumpAndSettle();
+        checkDropDownValue();
+
+        controller.setResponseViewType = NetworkResponseViewType.auto;
+        await tester.pumpAndSettle();
+        checkDropDownValue();
+
+        controller.stopRecording();
+
+        // pumpAndSettle so residual http timers can clear.
+        await tester.pumpAndSettle(const Duration(seconds: 1));
+      });
+
+      testWidgets(
+          'onChanged handler should trigger when changing drop down value manually',
+          (tester) async {
+        final currentResponseViewType = controller.currentResponseViewType;
+        final requestsNotifier = controller.requests;
+        await controller.startRecording();
+
+        // Load the network request.
+        await controller.networkService.refreshNetworkData();
+        expect(requestsNotifier.value.requests.length, equals(1));
+
+        // Select the request in the network request list.
+        final data = requestsNotifier.value.requests.first;
```

Comment:
same comment here - we don't need to do anything with the network controller. Just pump the HttpResponseTrailingDropDown widget directly, and you can pass some on changed handler that you'll be able to verify.

something like:
```dart
var selected = '';

...
onChanged: (value) {
  selected = value;
}
...

expect(selected, <the expected text value>);
```